pyqir-tests/tests/test_teleport.py

```diff
@@ -0,0 +1,143 @@
+from typing import Optional
+from pyqir.generator.module import SimpleModule
+from pyqir.generator.qis import BasicQisBuilder
+from pyqir.generator.value import Qubit
+from pyqir.jit.nonadaptivejit import NonadaptiveJit
+from pyqir.jit.gatelogger import GateLogger
+from pyqir.jit.gateset import GateSet
+import tempfile
+import unittest
+
+
+def teleport(qis: BasicQisBuilder, qubits: list, results: list) -> None:
+    msg = qubits[0]
+    target = qubits[1]
+    register = qubits[2]
+
+    # Create some entanglement that we can use to send our message.
+    qis.h(register)
+    qis.cx(register, target)
+
+    # Encode the message into the entangled pair.
+    qis.cx(msg, register)
+    qis.h(msg)
+
+    # Measure the qubits to extract the classical data we need to
+    # decode the message by applying the corrections on
+    # the target qubit accordingly.
+    # We use MResetZ from the Microsoft.Quantum.Measurement namespace
+    # to reset our qubits as we go.
+    qis.m(msg, results[0])
+    qis.reset(msg)
+    qis.if_result(results[0], one=lambda: qis.z(target))
+
+    # We can also use library functions such as IsResultOne to write
+    # out correction steps. This is especially helpful when composing
+    # conditionals with other functions and operations, or with partial
+    # application.
```

Comment:
These lines too are specific to the Q# style invocation, and not really to this version.